presto-main/src/main/java/io/prestosql/operator/HashAggregationOperator.java

```diff
@@ -368,6 +370,10 @@ public void addInput(Page page)
         if (aggregationBuilder == null) {
             // TODO: We ignore spillEnabled here if any aggregate has ORDER BY clause or DISTINCT because they are not yet implemented for spilling.
             if (step.isOutputPartial() || !spillEnabled || hasOrderBy() || hasDistinct()) {
+                aggregationBuilderMemoryContext = operatorContext.localUserMemoryContext();
+                if (useSystemMemory) {
+                    aggregationBuilderMemoryContext = operatorContext.localSystemMemoryContext();
```

Comment:
Why not pass MemoryContext into `InMemoryHashAggregationBuilder`?

Or, maybe better, why not have this assignment in the constructor?
As it is now, it looks like creating a Mem Context multiple times (which i think is not really the case).

Reply:
> Why not pass MemoryContext into InMemoryHashAggregationBuilder?

I tried that approach, but since `InMemoryHashAggregationBuilder` is used within `SpillableHashAggregationBuilder` it cannot account memory on it's own. This is because `SpillableHashAggregationBuilder` changes memory pools from revocable to user and `InMemoryHashAggregationBuilder` doesn't (and shouldn't) know about this flip.
Ideally (TODO), `SpillableHashAggregationBuilder` should use underlying accumulators and `GroupByHash` directly and omit `InMemoryHashAggregationBuilder`.

> Or, maybe better, why not have this assignment in the constructor?

That makes sense.